packages/mobile/src/identity/commentEncryption.ts

```diff
@@ -0,0 +1,309 @@
+// Wallet utilities for comment encryption and comment key management
+// Use these instead of the functions in @celo/utils/src/commentEncryption
+// because these manage comment metadata
+
+import { AccountsWrapper } from '@celo/contractkit/lib/wrappers/Accounts'
+import { IdentifierLookupResult } from '@celo/contractkit/lib/wrappers/Attestations'
+import { bufferToHex, eqAddress, hexToBuffer } from '@celo/utils/src/address'
+import {
+  decryptComment as decryptCommentRaw,
+  deriveCEK,
+  encryptComment as encryptCommentRaw,
+} from '@celo/utils/src/commentEncryption'
+import { getPhoneHash } from '@celo/utils/src/phoneNumbers'
+import { memoize, values } from 'lodash'
+import { call, put, select } from 'redux-saga/effects'
+import { TokenTransactionType, TransactionFeedFragment } from 'src/apollo/types'
+import { features } from 'src/flags'
+import { updateE164PhoneNumberAddresses, updateE164PhoneNumberSalts } from 'src/identity/actions'
+import {
+  getAddressesFromLookupResult,
+  lookupAttestationIdentifiers,
+} from 'src/identity/contactMapping'
+import { getUserSelfPhoneHashDetails, PhoneNumberHashDetails } from 'src/identity/privacy'
+import {
+  AddressToE164NumberType,
+  e164NumberToAddressSelector,
+  E164NumberToAddressType,
+  e164NumberToSaltSelector,
+  E164NumberToSaltType,
+} from 'src/identity/reducer'
+import { NewTransactionsInFeedAction } from 'src/transactions/actions'
+import Logger from 'src/utils/Logger'
+import { setPrivateCommentKey } from 'src/web3/actions'
+import { getContractKit } from 'src/web3/contracts'
+import { privateCommentKeySelector } from 'src/web3/selectors'
+
+const TAG = 'identity/commentKey'
+// A separator to split the comment content from the metadata
+const METADATA_CONTENT_SEPARATOR = '~'
+// Format should be separator + e164Number + salt
+const PHONE_METADATA_REGEX = new RegExp(
+  `(.*)${METADATA_CONTENT_SEPARATOR}([+][1-9][0-9]{1,14})([a-zA-Z0-9+/]{13})$`
+)
+
+// Derive a new comment key from the provided private key
+export function* createCommentKey(seedPrivateKey: string) {
+  Logger.debug(TAG, 'Creating a new comment key')
+  const privateCEK = bufferToHex(deriveCEK(seedPrivateKey))
+  yield put(setPrivateCommentKey(privateCEK))
+}
+
+export function* getCommentKey(address: string) {
+  const contractKit = yield call(getContractKit)
+  const accountsWrapper: AccountsWrapper = yield call([
+    contractKit.contracts,
+    contractKit.contracts.getAccounts,
+  ])
+  const hexString: string = yield call(accountsWrapper.getDataEncryptionKey, address)
+  return !hexString ? null : hexToBuffer(hexString)
+}
+
+export function* encryptComment(
+  comment: string | null,
+  toAddress: string | null,
+  fromAddress: string | null,
+  includePhoneNumMetadata: boolean = false
+) {
+  Logger.debug(TAG + 'encryptComment', 'Encrypting comment')
+  if (!features.USE_COMMENT_ENCRYPTION || !comment || !toAddress || !fromAddress) {
```

Comment:
I believe you've already raised this, but this is what you meant when you said comment encryption was only possible between two verified accounts, right?

Reply:
Currently, yes that's right. We should explore options for bringing this to more users